core-backend/sandbox/src/env.rs

```diff
@@ -159,14 +160,18 @@ impl<E: Ext + Into<ExtInfo> + 'static> Environment<E> for SandboxEnvironment<E>
         let info: ExtInfo = runtime.ext.unset().into();
 
         let termination = if res.is_err() {
-            let mut reason = None;
-
-            if let Some(trap) = runtime.trap {
-                if common_funcs::is_wait_trap(trap) {
-                    reason = Some(TerminationReason::Manual { wait: true });
+            let reason = if let Some(trap) = runtime.trap {
+                if let Some(value_dest) = info.exit_argument {
```

Comment:
I would place here something like `common::is_exit_trap` just to keep the flow. It also may cause some issues in future, if we decide to remove argument from `gr_exit`